bundle/edu.gemini.ags/src/main/scala/edu/gemini/ags/gems/GemsVoTableCatalog.scala

```diff
@@ -130,10 +131,27 @@ case class GemsVoTableCatalog(backend: VoTableBackend = RemoteBackend) {
   protected [gems] def getRadiusLimits(inst: GemsInstrument, criterions: List[GemsCatalogSearchCriterion]): List[RadiusConstraint] = {
     inst match {
       case GemsInstrument.flamingos2 => criterions.map(_.criterion.adjustedLimits)
-      case _                         => List(GemsUtils4Java.optimizeRadiusConstraint(criterions.asJava))
+      case _                         => List(optimizeRadiusConstraint(criterions.asJava))
     }
   }
 
+  // Combines multiple radius limits into one
+  protected [gems] def optimizeRadiusConstraint(criterList: java.util.List[GemsCatalogSearchCriterion]): RadiusConstraint = {
+    val result = criterList.asScala.foldLeft((Double.MinValue, Double.MaxValue)) { (prev, current) =>
+      val c = current.criterion
+      val radiusConstraint = c.adjustedLimits
+      val maxLimit = radiusConstraint.maxLimit
+      val correctedMax = (c.offset |@| c.posAngle) { (o, _) =>
+          // If an offset and pos angle were defined, normally an adjusted base position
+          // would be used, however since we are merging queries here, use the original
+          // base position and adjust the radius limits
+          maxLimit + o.distance
+        } | maxLimit
+      (max(correctedMax.toDegrees, prev._1), min(radiusConstraint.minLimit.toDegrees, prev._2))
+    }
+    RadiusConstraint.between(Angle.fromDegrees(result._1), Angle.fromDegrees(result._2))
```

Comment:
If the `criterList` is empty (which may not be possible?) this will be a strange constraint.  I'm not sure what `Double.MinValue` or `Double.MaxValue` would be when normalized to [0, 360) but probably nothing sensible.  Maybe we should return an empty constraint in that case?